addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3915,50 +3915,66 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
      * @param {function} computeShapeData function to compute the new shape data.
      */
     _handlePreviewState(previewMode, computeShapeData) {
+        const target = this.$target[0];
+        const insertShapeContainer = newContainer => {
+            const shapeContainer = target.querySelector(':scope > .o_we_shape');
+            if (shapeContainer) {
+                shapeContainer.remove();
+            }
+            if (newContainer) {
+                const preShapeLayerElement = this._getLastPreShapeLayerElement();
+                if (preShapeLayerElement) {
+                    preShapeLayerElement.after(newContainer);
+                } else {
+                    target.prepend(newContainer);
+                }
+            }
+            return newContainer;
+        };
+
+        let changedShape = false;
         if (previewMode === 'reset') {
-            if (this.prevShape) {
-                this.$target[0].dataset.oeShapeData = this.prevShape;
+            insertShapeContainer(this.prevShapeContainer);
+            if (this.prevShapeData) {
+                target.dataset.oeShapeData = this.prevShapeData;
```

Comment:
Seems like you again renamed a variable a customization might use, in a stable version... :)